.github/workflows/build-and-test.yml

```diff
@@ -75,15 +104,22 @@ jobs:
         uses: actions-rs/cargo@v1
         with:
           # Build the library, tests, and examples without running them to avoid recompilation in the run tests step
-          command: test
-          args: --all --release --no-run
+          command: build
+          args: --workspace --tests --examples --release
 
       - name: Run tests
         uses: actions-rs/cargo@v1
         with:
           command: test
           args: --all --release
 
+      - name: Run Rust examples
+        # run examples only on ubuntu for now
+        if: matrix.os == 'ubuntu-latest'
+        run: |
+          cargo read-manifest --manifest-path ./examples/Cargo.toml > ./example-manifest.json
+          # run examples in parallel with 4 threads and 1 retry
+          cat ./example-manifest.json | jq -r '.targets[].name' | parallel -k -j 4 --retries 1 cargo run --example {} --release
```

Comment:
Nit: do we still need the intermediate file? If the concern is readability could we pipe these over separate lines?

Super optional, I'm totally fine leaving this as-is and don't know if there are any caveats with the trailing pipes.

E.g.
```bash
          # Run examples in parallel with 4 threads and 1 retry.
          cargo read-manifest --manifest-path ./examples/Cargo.toml |
          cat ./example-manifest.json |
          jq -r '.targets[].name' | 
          parallel -k -j 4 --retries 1 cargo run --example {} --release
```